### ISSUE TITLE
Segmentation fault during compilation

### ISSUE THREAD

**Issue by georgii-ivanov** [vercel/next.js#23045]
### What version of Next.js are you using?

10.0.8

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying your application?

next build

### Describe the Bug

$ next build
info  - Using external babel configuration from /var/www/project/babel.config.json
@rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to `true`, as the next major version will default this option to `true`.
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
info  - Generating static pages (3/3)
info  - Finalizing page optimization  

Segmentation fault (core dumped)
error Command failed with exit code 139.

Linux specs: > 32gb RAM, 8 cores/16 threads, > 400gb free space

### Expected Behavior

$ next build
info  - Using external babel configuration from /var/www/project/babel.config.json
@rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to `true`, as the next major version will default this option to `true`.
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
Page                                                           Size     First Load JS
┌ λ /                                                          2.59 kB         301 kB
├   /_app                                                      0 B             201 kB
├ λ /404                                                       407 B           299 kB
├ λ /blog                                                      2.85 kB         301 kB
├ λ /blog/add                                                  779 B           310 kB
├ λ /blog/edit                                                 955 B           310 kB
├ λ /blog/view                                                 4.79 kB         307 kB
├ λ /forgot                                                    1.99 kB         304 kB
├ λ /login                                                     2.21 kB         304 kB
├ λ /logout                                                    591 B           299 kB
├ λ /register                                                  2.13 kB         304 kB
├ λ /reset                                                     2.06 kB         304 kB
+ First Load JS shared by all                                  201 kB
  ├ chunks/6b4f7fe54e590bcf446803a5ab72bd111bc8cff9.4f9987.js  10.9 kB
  ├ chunks/6d26015278f57e4bd3218213751cc4809bf22e52.abb295.js  8.79 kB
  ├ chunks/commons.0be295.js                                   118 kB
  ├ chunks/framework.728d31.js                                 46.5 kB
  ├ chunks/main.f98c7d.js                                      6.36 kB
  ├ chunks/pages/_app.2789fd.js                                8.41 kB
  └ chunks/webpack.397c58.js                                   1.22 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
   (ISR)     incremental static regeneration (uses revalidate in getStaticProps)

### To Reproduce

Run `next build`

This doesn't fail on my Mac workstation but fails on my production Linux.

It's not reproducible on all old nextjs versions (tried 10.0.7) already.

Let me know what logs will you need from me.

**Comment by georgii-ivanov** [comment]
It's now resolved with a new next.js version.
Self-fixed issue. You guys are so quick!!!
Closing it...

**Comment by dryleaf** [comment]
I was wondering how to address the warning, `@rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to true, as the next major version will default this option to true.`, which is displayed in the issue description.

**Comment by dayvista** [comment]
@georgii-ivanov what version fixed this issue for you? I'm assuming `10.0.9`.

I'm getting this error locally (running `next build` on Linux Ubuntu 20.04).
When deploying on Vercel, the build fails at the same step, though with a different message:

```node
22:36:26.608 | info  - Generating static pages (10/10)
-- | --
22:36:26.609 | info  - Finalizing page optimization...
22:36:29.441 | Error: Command "npm run build" exited with SIGSEGV
22:36:30.512 | [  247.683213] fargate-bridge: port 1(vethce8213b5) entered forwarding state
22:36:30.512 | [  247.741864] ena 0000:00:06.0 eth1: Local page cache is disabled for less than 16 channels
22:36:30.512 | [  311.215760] cgroup: cgroup: disabling cgroup2 socket matching due to net_prio or net_cls activation
22:36:30.512 | [  583.164881] node[6073]: segfault at 7fca4c7cae00 ip 00007fca4c7cae00 sp 00007fca4e1fbe48 error 14 in libz.so.1[7fca4c9ce000+17000]
```

The error initially showed up while using Next.js version `10.0.7`. It persists after upgrading to `10.0.9` and `10.2.0`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
